app/src/main/java/org/astraea/performance/Manager.java

```diff
@@ -113,25 +121,25 @@ public boolean consumedDone() {
   private static class RandomContent {
     private final Random rand = new Random();
     private final DataSize dataSize;
-    private final boolean fixedSize;
+    private final Distribution distribution;
     private final byte[] content;
 
     /**
      * @param dataSize The size of each random generated content in bytes.
-     * @param fixedSize Determine whether to fix the size of random generated content
+     * @param distribution Determine whether to fix the size of random generated content or random
+     *     size with specified distribution
      */
-    public RandomContent(DataSize dataSize, boolean fixedSize) {
+    public RandomContent(DataSize dataSize, Distribution distribution) {
       this.dataSize = dataSize;
-      this.fixedSize = fixedSize;
+      this.distribution = distribution;
       content = new byte[dataSize.measurement(DataUnit.Byte).intValue()];
     }
 
     public byte[] getContent() {
       // Randomly change one position of the content;
       content[rand.nextInt(dataSize.measurement(DataUnit.Byte).intValue())] =
           (byte) rand.nextInt(256);
```

Comment:
The generated data should have following essential attributes.
1. random data
2. random size

You do the fully copy here, so it should be fine to create new byte array and fill the data by `rand.nextBytes`

Reply:
> You do the fully copy here, so it should be fine to create new byte array and fill the data by `rand.nextBytes`

If we fill the data by `rand.nextBytes`, the throughput of producers goes down. (~1GiB -> ~70MiB)
I think that the bottleneck is `rand.nextBytes`. Jvm takes a lot of time calculating random value of **each** bytes, but takes a short time on copying array.

> 1. random data

Considering the calculation time, the way I achieve "random data" is **change one byte each time** I call `getContent`. I'll keep thinking if there is any efficient way to generate "more random" data.